src/blockdigest.cpp

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) 2017 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include "blockdigest.h"
+#include "streams.h"
+#include "version.h"
+#include "uint256.h"
+#include "primitives/transaction.h"
+#include "primitives/block.h"
+#include "validation.h"
+#include "chainparams.h"
+#include "utiltime.h"
+
+namespace DigestFilter {
+
+data_t base_filter::outpoint_data(const COutPoint& outpoint) const
+{
+    CDataStream stream(SER_NETWORK, PROTOCOL_VERSION);
+    stream << outpoint;
+    data_t data(stream.begin(), stream.end());
+    return data;
+}
+
+data_t base_filter::hash_data(const uint256& hash) const
```

Comment:
Get rid of this. Just change this to be tx_data.emplace_back(hash.begin(), hash.end()) inline.